packages/cli/src/commands/hydrogen/env/push__unstable.ts

```diff
@@ -0,0 +1,243 @@
+import {diffLines} from 'diff';
+import {commonFlags, flagsToCamelObject} from '../../../lib/flags.js';
+import {login} from '../../../lib/auth.js';
+import {getCliCommand} from '../../../lib/shell.js';
+import {resolvePath} from '@shopify/cli-kit/node/path';
+import {
+  renderConfirmationPrompt,
+  renderSelectPrompt,
+  renderInfo,
+  renderWarning,
+  renderSuccess,
+  renderError,
+} from '@shopify/cli-kit/node/ui';
+import {
+  loadEnvironmentVariableFile,
+  parseDotEnvFile,
+} from '@shopify/oxygen-cli/utils';
+import {fileExists, readFile} from '@shopify/cli-kit/node/fs';
+import {
+  outputContent,
+  outputToken,
+} from '@shopify/cli-kit/node/output';
+import {
+  renderMissingLink,
+} from '../../../lib/render-errors.js';
+import {Environment, getStorefrontEnvironments} from '../../../lib/graphql/admin/list-environments.js';
+import {linkStorefront} from '../link.js';
+import {EnvironmentVariable, getStorefrontEnvVariables} from '../../../lib/graphql/admin/pull-variables.js';
+import {HydrogenStorefrontEnvironmentVariableInput, pushStorefrontEnvVariables} from '../../../lib/graphql/admin/push-variables.js';
+import { AbortError } from '@shopify/cli-kit/node/error';
+
+export default class EnvPush extends Command {
+  static description =
+    'Push your local variables from .env to your Hydrogen storefront.';
+
+  static flags = {
+    environment: commonFlags.environment,
+    path: commonFlags.path,
+  };
+
+  async run(): Promise<void> {
+    const {flags} = await this.parse(EnvPush);
+    await runEnvPush({...flagsToCamelObject(flags)});
+  }
+}
+
+interface Flags {
+  environment?: string;
+  path?: string;
+}
+
+export async function runEnvPush({
+  environment,
+  path = process.cwd(),
+}: Flags) {
+  let validatedEnvironment: Partial<Environment> = {};
+
+  // Ensure local .env file
+  const dotEnvPath = resolvePath(path, '.env');
+  let localVariables: Record<string, string>;
+
+  try {
+    localVariables = parseDotEnvFile(dotEnvPath);
+  } catch {
+    renderWarning({
+      body: 'Local .env file not found'
+    });
+    process.exit(1);
+  }
+
+  // Authenticate
+  const [{session, config}, cliCommand] = await Promise.all([
+    login(path),
+    getCliCommand(),
+  ]);
+
+  if (!config.storefront?.id) {
+    renderMissingLink({session, cliCommand});
+
+    const runLink = await renderConfirmationPrompt({
+      message: outputContent`Run ${outputToken.genericShellCommand(
+        `${cliCommand} link`,
+      )}?`.value,
+    });
+
+    if (!runLink) return;
+
+    config.storefront = await linkStorefront(path, session, config, {
+      cliCommand,
+    });
+  }
+
+  if (!config.storefront?.id) return;
+
+  // Fetch environments
+  const {environments: environmentsData} = await getStorefrontEnvironments(
+    session,
+    config.storefront.id,
+  ) ?? {};
+
+  if (!environmentsData) {
+    renderWarning({
+      body: 'Failed to fetch environments',
+    });
+    process.exit(1);
```

Comment:
Similar story here, `throw new AbortError(...)` should be preferred if this can be considered an unexpected error. -- and for all the other warning + process.exit in the code